src/speechsynthesizer/SAPISpeechSynthesizer.cpp

```diff
@@ -31,19 +32,24 @@ void SAPISpeechSynthesizer::DoUninitialize() {
 }
 
 void SpeakThreadTask(const char* text, const char* language) {
-    auto speak =
-        fmt::format("<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' xml:lang='{}'>{}</speak>",
-                    language, text)
-            .c_str();
+    const size_t languageSize = strlen(language) + 1;
+    wchar_t* wLanguage = new wchar_t[languageSize];
+    mbstowcs(wLanguage, language, languageSize);
 
-    const int w = 512;
-    int* wp = const_cast<int*>(&w);
-    *wp = strlen(speak);
+    std::string locale = fmt::format("{}.UTF-8", language);
+    locale.replace(locale.find("-"), 1, "_");
+    std::setlocale(LC_ALL, locale.c_str());
 
-    wchar_t wtext[w];
-    mbstowcs(wtext, speak, strlen(speak) + 1);
+    const size_t textSize = strlen(text) + 1;
+    wchar_t* wText = new wchar_t[textSize];
+    mbstowcs(wText, text, textSize);
 
-    mVoice->Speak(wtext, SPF_IS_XML | SPF_ASYNC | SPF_PURGEBEFORESPEAK, NULL);
+    auto speakText = fmt::format(
+        L"<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' xml:lang='{}'>{}</speak>", wLanguage, wText);
+    mVoice->Speak(speakText.c_str(), SPF_IS_XML | SPF_ASYNC | SPF_PURGEBEFORESPEAK, NULL);
```

Comment:
What happens to the memory allocated into `textSize` and `wLanguage` i this function, or any other, throws an exception? I guess for now it doesn't matter because the only catch is going to be in the crash handler. Just wanted to point this out.

Reply:
It's actually less about whether or not it does throw an exception and more that someone could add code which could throw an exception and thus cause a memory leak. :)

std::wstring should alleviate this concern.